at-quannguyen/W4/ex3.rb

```diff
@@ -0,0 +1,32 @@
+class ArrayClone
+  attr_accessor :array, :length
+  def initialize
+    self.array = []
+    self.length = 0
+  end
+
+  def map(&block)
+    result = []
+    @array.each { |e| result << block.(e) }
+    result
+  end
+
+  def select(&block)
+    result = []
+    @array.each { |e| result << i if block.(e) }
+    result
+  end
+
+  def length
+    @array.each { |e| @length += 1 }
+    @length
+  end
+end
+
+arr = ArrayClone.new
+arr.array = [1, 2, 3, 5, 6]
+
+p arr.array.length
+new_arr = arr.map { |i| i + 1 }
+p new_arr
+p arr.select { |i| i > 3 }
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/atruby-team/internship-18-summer/blob/cd24ce24c04c57d4b75cc24b8a37d518f57e5e0e/at-quannguyen/W4/ex3.rb#L32"></span>Lint/AmbiguousBlockAssociation: Parenthesize the param <code>arr.select { |i| i &gt; 3 }</code> to make sure that the block will be associated with the <code>arr.select</code> method call. (https://github.com/rubocop-hq/ruby-style-guide#syntax)</td>
    </tr>
  </tbody>
</table>